bestiary/core/models.py

```diff
@@ -178,6 +180,42 @@ def __str__(self):
         return '%s - %s - %s - %s - %s' % (self.ouid, self.trx, self.op_type, self.entity_type, self.target)
 
 
+class Project(EntityBase):
+    """Model class for Project objects.
+
+    This class is meant to represent an abstract set of end-point
+    views sharing a common context. The projects can also have n
+    sub-projects which also have sets of end-point views. A project
+    could belong to one single ecosystem, to different ones or not
+    belonging to any ecosystem.
+
+    Every project object must have a name (`name`).
+    Optionally, it may have a title (`title`), a parent project
+    (`parent_project`) and a related ecosystem (`ecosystem`).
+
+    :param name: Name of the project
+
+    """
+    name = CharField(unique=True,
+                     max_length=MAX_SIZE_NAME_FIELD,
+                     help_text='Project name')
+    title = CharField(max_length=MAX_SIZE_TITLE_FIELD,
+                      null=True,
+                      help_text='Project title')
+    parent_project = ForeignKey("Project",
+                                parent_link=True,
+                                null=True,
+                                on_delete=SET_NULL)
```

Comment:
Maybe it's better if this is set to `ON DELETE CASCADE`, so child projects will be deleted automatically.

Reply:
I did this on purpose, as I thought maybe the user would like to remove only a "child" project but not its parent (and if the parent has a parent too, it also will be deleted). I'm changing the behavior then!

Reply:
If you remove the children, you don't remove the parent. `on_delete` works when you delete the foreign key, in this case, the parent.

Reply:
Oops! You are right, sorry for the confusion.